src/cmap/connection.ts

```diff
@@ -125,6 +126,8 @@ export interface ConnectionOptions
   extendedMetadata: Promise<Document>;
   /** @internal */
   mongoLogger?: MongoLogger | undefined;
+  /** @internal */
+  monitor?: Monitor;
```

Comment:
This is not exactly how I pictured this, because now pooled connections will need to be given access to the monitor of the server that they come from rather than the pool or server which is a more common shared structure. I think we should have less of a strict binding of exactly what type is needed here, connections may want to have a `parent` that fits a common type contract as little as `{server: Server}` or less `{serverDescription: ServerDescription}`, then monitor, pool, or anything else needs to provide a getter to satisfy that. 

We can skip working on this in this PR, I figured it fit because where we ended up storing the relvant information affected the way we obtained it. I think now that it is clear the serverDescription provides the RTT data we can figure out how to provide that later.

Reply:
I'll remove this from this PR then.